bundler/spec/dependabot/bundler/helper_spec.rb

```diff
@@ -41,7 +41,7 @@
   end
 
   let(:v1) { "1.17.3" }
-  let(:v2) { "2.3.8" }
+  let(:v2) { "2.3.9" }
```

Comment:
I think this spec could be relaxed to not hardcode exact versions, but instead just check the major version selected.

Reply:
I'm not sure, right now I think the exact version is relevant, if we do [this](https://github.com/dependabot/dependabot-core/pull/4884#discussion_r830971299) then we can maybe relax it?

Reply:
These specs are unit tests for the `Bundler::Helpers.bundler_version` method, which selects an exact bundler version according to the lockfile contents, using the following criteria:

* No lockfile -> The exact v2 version dependabot has available.
* Lockfile with no BUNDLED WITH section -> The exact v2 version dependabot has available.
* Lockfile with BUNDLED WITH 1.x section -> The exact v1 version dependabot has available.
* Lockfie with BUNDLED WITH 2.x section or higher -> The exact v2 version dependabot has available.

Namely, a unit test for this method

https://github.com/dependabot/dependabot-core/blob/ea8fbaf64af5af477d5b69d0b985782340ca3b36/bundler/lib/dependabot/bundler/helpers.rb#L18-L26

In my opinion, testing that the result is a valid 3 component version number, and that the first segment is correct (1.x or 2.x) is enough. The exact version itself that dependabot is using is "configuration" and should not need a test, because it's not useful for detecting bugs and it means essentially testing the specific value of a constant, which defeats one of the points of using constants (keep things DRY).

Reply:
That's said that just my personal opinion but it's no big deal, and indeed if https://github.com/dependabot/dependabot-core/pull/4884#discussion_r830971299 works, all this code will go away, so these tests will go too anways.